testutil/garbage/garbage.go

```diff
@@ -0,0 +1,304 @@
+package garbage
+
+import (
+	"math"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipld/go-ipld-prime/datamodel"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	"github.com/ipld/go-ipld-prime/must"
+	basicnode "github.com/ipld/go-ipld-prime/node/basic"
+	"github.com/multiformats/go-multihash"
+)
+
+type Options struct {
+	initialWeights map[datamodel.Kind]int
+	weights        map[datamodel.Kind]int
+	blockSize      uint64
+}
+
+type generator func(count uint64, opts Options) (uint64, datamodel.Node)
+
+type hasher struct {
+	code   uint64
+	length int
+}
+
+const charset = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789`~!@#$%^&*()-_=+[]{}|\\:;'\",.<>?/ \t\n☺💩"
+
+var codecs []uint64
+var hashes []hasher
+var kinds datamodel.KindSet
+var generators map[datamodel.Kind]generator
+var runes []rune
+
+// Garbage produces random Nodes which can be useful for testing and benchmarking. By default, the
+// Nodes produced are relatively small, averaging near the 100 byte range when encoded
+// (very roughly, with a wide spread).
+//
+// Options can be used to adjust the average size and weights of occurances of different kinds
+// within the complete Node graph.
+func Garbage(opts ...Option) datamodel.Node {
+	options := applyOptions(opts...)
+	rand.Seed(time.Now().UnixNano())
+	_, n := generate(options.blockSize, options)
+	return n
+}
+
+func generate(count uint64, opts Options) (uint64, datamodel.Node) {
+	weights := opts.weights
+	if opts.initialWeights != nil {
+		weights = opts.initialWeights
+		opts = Options{weights: opts.weights}
+	}
+	totWeight := 0
+	for _, kind := range kinds {
+		totWeight += weights[kind]
+	}
+	r := rand.Float64() * float64(totWeight)
+	var wacc int
+	for _, kind := range kinds {
+		wacc += weights[kind]
+		if float64(wacc) >= r {
+			return generators[kind](count, opts)
+		}
+	}
+	panic("bad options")
+}
+
+func rndSize(bias uint64) uint64 {
+	// @rv: OK, I'll admit that I've completely forgotten how I arrived at this, but it
+	// generates numbers that bias toward `bias`, but not strictly, they may vary wildly
+	// but be drawn toward the lower end of the range and tend to be suitable for lengths
+	// of arrays and such
+	m := math.Abs(math.Floor(5/(1-(math.Pow(rand.Float64(), 2))) - 5 + rand.Float64()*float64(bias)))
```

Comment:
How about https://pkg.go.dev/math/rand#NormFloat64 instead?

It gives you a normal distribution, and the docs show how you can combine that with your desired mean and standard deviation. Plus, you can easily hook it up to a user-provided random source, via rand.Rand.

Reply:
yeah, that's a neat API, have used it instead